### PR TITLE
API Deprecation - fix 2381

### LINKED ISSUE #2381: [API Deprecation]: ParameterGroup, BuiltinParameterGroup, GetLabelFor, GetParameterGroupTypeId, GetBuiltInParameterGroup
### 🐞 Describe the bug

Need to review and change API deprecated methods:
- [x] [ParameterGroup](https://apidocs.co/apps/revit/2024/bc927bca-8fbe-9f2d-8acb-785f65c57969.htm) property of InternalDefinition, ExternalDefinition, and Definition
- [x] [BuiltinParameterGroup](https://apidocs.co/apps/revit/2024/9942b791-2892-0658-303e-abf99675c5a6.htm) enum
- [x] [GetLabelFor Method (BuiltInParameterGroup)](https://apidocs.co/apps/revit/2024/b48d806c-d7c5-7638-c6f8-041495c5d783.htm)
- [x] [GetParameterGroupTypeId](https://apidocs.co/apps/revit/2024/51728d1e-61d4-ecd3-5219-0dd007ec855c.htm) and [GetBuiltInParameterGroup](https://apidocs.co/apps/revit/2024/3703f1ea-a444-518c-7112-e9a1303dac12.htm)

Handle with:
```python
if HOST_APP.is_newer_than(2023):
    # new method
else:
    # old method
```

### Additional context

see https://discourse.pyrevitlabs.io/t/release-date-pyrevit-2025/3083/60?u=jean-marc
and https://github.com/pyrevitlabs/pyRevit/pull/2380 for similar handling

### PR DESCRIPTION
**Pull request by jmcouffin**
fixes #2381

### CODE REVIEW

File: extensions/pyRevitTools.extension/pyRevit.tab/Project.panel/Wipe.pulldown/Wipe All Project Parameters.pushbutton/script.py [hunk 1]
```diff
@@ -24,16 +24,17 @@
         BIND = 'Unknown'
 
     name = p.Name
-    if HOST_APP.is_newer_than('2022'):
-        ut = str(p.GetDataType().TypeId)
-        tp = str(p.GetDataType().TypeId.split('.')[-3])
-    elif HOST_APP.is_exactly('2022'):
+    if HOST_APP.is_exactly('2022'):
```
Comment (sanzoghenzo) [inline extensions/pyRevitTools.extension/pyRevit.tab/Project.panel/Wipe.pulldown/Wipe All Project Parameters.pushbutton/script.py:27]: this if branch doesn't set the pg variable

File: pyrevitlib/pyrevit/revit/db/__init__.py [hunk 2]
```diff
@@ -181,23 +181,21 @@ def __init__(self, param_def, param_binding=None, param_ext_def=False):
         # Revit <2017 does not have the Id parameter
         self.param_id = getattr(self.param_def, 'Id', None)
 
-
-        if HOST_APP.is_newer_than(2022, or_equal=True):
-            # GetSpecTypeId() Removed in Revit 2022
-            self.unit_type = self.param_def.GetDataType()
-        elif HOST_APP.is_exactly(2021):
+        if HOST_APP.is_exactly(2021):
```
Comment (sanzoghenzo) [inline pyrevitlib/pyrevit/revit/db/__init__.py:184]: this if branch doesn't set the param_type and param_group attributes

**Review by sanzoghenzo — COMMENTED** [review]
Unfortunately this fix introduces new bugs!
since you moved the variables and attributes inside the `else` branch, they should be set in all the other branches, too.

**Comment by sanzoghenzo** [comment]
@dosymep did you get any notification for that? the code is broken and it needs to be fixed before you do anything else :wink:

File: pyrevitlib/pyrevit/revit/db/__init__.py (hunk 2 above)
Comment (jmcouffin) [reply pyrevitlib/pyrevit/revit/db/__init__.py:184]: I was to prompt to make this PR on my laptop yesterday night, forgot the last two ocmmit before merging! thnaks for the catch

**Pull request merged**